### PR TITLE
CLI set up for running prediction using a previous model

### PR DESCRIPTION
**Pull request by ecpierce**
main changes:
-added notes to readme about what has changed and how to run
- made new scripts to run separate module to do prediction if you already have a model, and edited bsoid_app/predict.py to work with this

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,10 +62,23 @@ conda activate bsoid_v2
 
 You should now see (bsoid_v2) $yourusername@yourmachine ~ %
 
-#### Step 2: Run the app!
+#### Step 2: Run the pipeline through the CLI!
+##### See run_streamlit_cli.py and run_streamlit_cli_predict.py for explanations of the environmental variables that need to be set here and default settings where they exist.
```
Comment (taylorreiter) [inline README.md]: wait im confused. the user has to go into those python scripts to look at the documentation? would it be simple to put that in the readme? if not, can you at least link to the py files?

**Review by taylorreiter — COMMENTED** [review]
lgtm, a couple questions about the readme, but again will wait until Mert's approval! congrats on getting it this far!

File: bsoid_predict.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import os
+
+from streamlit import caching
+
+from bsoid_app import predict
+from bsoid_app.bsoid_utilities import visuals
+from bsoid_app.bsoid_utilities.load_workspace import *
+
+def get_env_variable(var_name, default_value=None):
```
Comment (mertcelebi) [inline bsoid_predict.py:9]: You could probably move this to a `utils.py` and import it in both places to reduce duplicaion

File: bsoid_predict.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+except ValueError as e:
+    print(e)
+    exit(1)  # Exit if any required variable is missing or if a conversion to float fails
+
+[ROOT_PATH, DATA_DIRECTORIES, FRAMERATE, pose_chosen, input_filenames, _, processed_input_data, _] \
+    = load_data(WORKING_DIR, PREFIX)
+[_, _, _, clf, _, predictions] = load_classifier(WORKING_DIR, PREFIX)
+
+predictor = predict.Prediction(ROOT_PATH, DATA_DIRECTORIES, input_filenames, processed_input_data, WORKING_DIR,
```
Comment (mertcelebi) [inline bsoid_predict.py]: You probably should use the ruff VSCode extension to lint these type of lines

File: bsoid_app/predict.py [hunk 4]
```diff
@@ -34,7 +31,7 @@ def __init__(self, ROOT_PATH, DATA_DIRECTORIES, input_filenames, processed_input
         self.use_train = []
         self.new_root_path = []
         self.new_directories = []
-        self.filetype = []
+        self.filetype = FTYPE
```
Comment (mertcelebi) [inline bsoid_app/predict.py:34]: `FTYPE` was an array but here is a string. Will this cause issues? Or do you need smth like `self.filetype = [FTYPE]`?

File: bsoid_app/predict.py [hunk 5]
```diff
@@ -47,173 +44,130 @@ def __init__(self, ROOT_PATH, DATA_DIRECTORIES, input_filenames, processed_input
...
-            d4 = today.strftime("%b-%d-%Y")
-            self.new_prefix = st.text_input('Enter new prediction variable PREFIX:', d4)
-            if self.new_prefix:
-                st.markdown('You have chosen **{}_predictions.sav** for new predictions.'.format(self.new_prefix))
-            else:
-                st.error('Please enter a name for your new prediction variable PREFIX.')
-        else:
-            self.new_prefix = self.prefix
+        self.new_root_path = self.root_path
+        self.filetype = [s for i, s in enumerate(['csv', 'h5', 'json'])
```
Comment (mertcelebi) [inline bsoid_app/predict.py]: Yeah line 34 won't work

**Review by mertcelebi — CHANGES_REQUESTED** [review]
Some small fixes

File: README.md (hunk 1 above)
Comment (ecpierce) [reply README.md]: added to readme

File: bsoid_predict.py (hunk 3 above)
Comment (ecpierce) [reply bsoid_predict.py]: reformatted for readability

File: bsoid_app/predict.py (hunk 4 above)
Comment (ecpierce) [reply bsoid_app/predict.py:34]: think fine as string, originally it is a dropdown selection, but there is only one ftype

File: bsoid_app/predict.py (hunk 5 above)
Comment (ecpierce) [reply bsoid_app/predict.py]: removed these lines, we don't need them. self.filetype is defined in setup

File: bsoid_predict.py (hunk 2 above)
Comment (ecpierce) [reply bsoid_predict.py:9]: opting not to do this

**Review by taylorreiter — COMMENTED** [review]
Just commenting that I have nothing to add...congrats on getting it this far!

**Review by mertcelebi — APPROVED** [review]

**Pull request merged**